coalaip/models.py

```diff
@@ -0,0 +1,380 @@
+                persistence layer
+            to_user (any): a user based on the model specified by the
+                persistence layer
+
+        Returns:
+        """
+
+        if self._persist_id is None:
+            raise EntityNotYetPersistedError(('Entities cannot be transferred '
+                                              'until they have been persisted'))
+        else:
+            return self._plugin.transfer(self._persist_id, transfer_payload,
+                                         from_user=from_user, to_user=to_user)
+
+
+class Creation(CoalaIpEntity):
+    """
+
+    INSTANTIATION NOTES
+    """
+
+    def __init__(self, data, *, entity_type='CreativeWork', plugin, **kwargs):
+        """Initialize a :class:`~coalaip.models.Creation` instance
+
+        INSTANTIATION NOTES
+        """
+
+        creation_name = data.get('name')
+
+        if not isinstance(creation_name, str):
+            # Proper error type... is ValueError right? Or KeyError? Or both?
+            raise EntityDataError(("'name' must be given as a string in the "
+                                   "'data' of Creations (Works and Manifestations). "
+                                   "Given '{}' instead.".format(creation_name)))
+
+        super().__init__(data, entity_type=entity_type, plugin=plugin, **kwargs)
+
+
+class Work(Creation):
+    """
+
+    INSTANTIATION NOTES
+    """
+
+    def __init__(self, data, *, plugin, **kwargs):
+        """Initialize a :class:`~coalaip.models.Work` instance
+
+        INSTANTIATION NOTES
+        """
+
+        if 'manifestationOfWork' in data:
+            raise EntityDataError(("'manifestationOfWork' must not be given "
+                                   "in the 'data' of Works"))
+        if data.get('isManifestation', False):
+            raise EntityDataError(("'isManifestation' must not be True if "
+                                   "given in the 'data' of Works"))
+
+        super().__init__(data, plugin=plugin, **kwargs)
+
+
+class Manifestation(Creation):
+    """
+
+    INSTANTIATION NOTES
+    """
+
+    def __init__(self, data, *, entity_type='CreativeWork', plugin, **kwargs):
+        """Initialize a :class:`~coalaip.models.Manifestation` instance
+
+        INSTANTIATION NOTES
+        """
+
+        manifestation_of = data.get('manifestationOfWork')
+
+        if not isinstance(manifestation_of, str):
+            raise EntityDataError(("'manifestationOfWork' must be given as a "
+                                   "string in the 'data' of Copyrights. "
+                                   "Given '{}' instead.".format(manifestation_of)))
+
+        # If the entity type is already specified as part of the data, use that
+        # instead of 'CreativeWork'
+        for type_key in ['type', '@type']:
+            if type_key in data:
+                entity_type = data[type_key]
+                del data[type_key]
+                break
+
+        # FIXME: apply defaults
```

Comment:
Nothing really comes to mind for defaults in Manifestations or Works, but maybe I missed something?